### PR TITLE
Upgrade terraform-provider-hcloud to v1.37.0

### LINKED ISSUE #178: Upgrade terraform-provider-hcloud to v1.37.0
Release details: https://github.com/hetznercloud/terraform-provider-hcloud/releases/tag/v1.37.0

### PR DESCRIPTION
**Pull request by aq17**
This PR was generated via `$ upgrade-provider pulumi-hcloud --migration-opts=autoalias`.

---

Upgrading terraform-provider-hcloud from 1.36.2 to 1.37.0.
Upgrading pulumi-terraform-bridge from v3.42.0 to v3.43.0.

Fixes #178

### CODE REVIEW

**Review by iwahbe — CHANGES_REQUESTED** [review]
This still has all the java artifacts. They should not be in this PR.

File: provider/resources.go [hunk 1]
```diff
@@ -16,22 +16,29 @@ package hcloud
...
 	"github.com/hetznercloud/terraform-provider-hcloud/hcloud"
 	"github.com/pulumi/pulumi-hcloud/provider/pkg/version"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/x"
 	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
+var
```
Comment (aq17) [inline provider/resources.go:32]: The token position is generated incorrectly from https://github.com/pulumi/upgrade-provider/blob/aqiu/code-migration-ast/migrations/migrations.go#L43-L50

File: provider/resources.go [hunk 2]
```diff
@@ -16,22 +16,29 @@ package hcloud
 
 import (
 	"fmt"
+	_ "embed"
```
Comment (aq17) [inline provider/resources.go:19]: The required comment explaining blank imports is not being added from https://github.com/pulumi/upgrade-provider/blob/aqiu/code-migration-ast/migrations/migrations.go#L29-L33

Comment (iwahbe) [reply provider/resources.go:19]: I don't think we actually need this comment. We can assume that go programmers understand what `_ "${PKG}"` means.

Comment (aq17) [reply provider/resources.go:19]: The lint test will complain and fail though, unless we're okay with ignoring that

Comment (iwahbe) [reply provider/resources.go:19]: Ok. Then we need to keep it.

**Pull request closed without merging**